Producer/Producer.Shared/Auth/Keychain.cs

```diff
@@ -123,9 +123,12 @@ KeyStore getKeystore (string service)
 					keystore.Load (stream, password);
 				}
 			}
-			catch (FileNotFoundException)
+			catch (Exception e)
 			{
-				keystore.Load (null, password);
+				if (e is FileNotFoundException || e is EOFException)
+					keystore.Load (null, password);
+				else
+					throw e;
```

Comment:
`throw e` should be `throw` (remove the e)

There is a difference - most of the time you just want to use `throw`
- `throw e` resets the stack trace (so your errors would appear to originate from getKeystore)
- `throw` doesn't - the original offender would be preserved (whatever failed internally)

Reply:
removed e